### PR TITLE
[WIP] Added hero field for adding custom background images

### PR DESCRIPTION
**Pull request by Joseph5062A**

### CODE REVIEW

**Comment by andy-wojciechowski** [comment]
To fix the failing tests you will need to mock the method that you made in the PostService, is_valid_hero() to return false. If you need an example look for uses of expects in the tests.

Also, please run rubocop by using `bundle exec rubocop` and fix any offenses. It's a stylecheck which runs against your code which is hooked up to travis. I know it's like Checkstyle but running those checks is very important for a language like Ruby.

File: app/controllers/post_controller.rb [hunk 1]
```diff
@@ -41,14 +42,18 @@ def submit
...
+    def validate_submission_parameters(title, author, markdown_text, hero)
       validation_message = nil
       if title.empty?
         validation_message = 'A post cannot be submited with a blank title.'
       elsif author.empty?
         validation_message = 'A post cannot be submited without an author.'
       elsif markdown_text.empty?
-        validation_message = 'A post cannot be submited with no markdown content.'
+        validation_message = 'A post cannot be submited with no markdown content.' 
+      elsif !hero.empty? && !(hero =~ URI::regexp)
```
Comment (andy-wojciechowski) [inline app/controllers/post_controller.rb]: Please add a test case here for the submit method for an invalid URL. 

The test block should be something like `test 'post/submit should redirect back to the edit screen with an error message if a post was submited with a hero that's an invalid URL.'`

If you need an example of what you're test should look like. See the test block `test 'post/submit should redirect back to the edit screen with an error message if a post was submited without markdown'` in post_controller_test.rb

File: app/controllers/post_controller.rb [hunk 2]
```diff
@@ -41,14 +42,18 @@ def submit
...
+        validation_message = 'The background image must be a valid URL.'
+      elsif !PostService.is_valid_hero(hero)
```
Comment (andy-wojciechowski) [inline app/controllers/post_controller.rb]: Same thing like above. Please add a test case for when the hero url is not image.

File: app/services/kramdown_service.rb [hunk 3]
```diff
@@ -63,22 +63,24 @@ def get_image_filename_from_markdown(image_file_name, markdown)
...
       parsed_tags = parse_tags(tags)
 
       tag_section = %(tags:
 #{parsed_tags})
       
       lead_break_section = "{: .lead}\r\n<!–-break-–>"
-          
+      
+      hero_to_use = hero
+      hero_to_use = 'https://source.unsplash.com/collection/145103/' if hero_to_use.empty?
```
Comment (andy-wojciechowski) [inline app/services/kramdown_service.rb]: Please add a test case for when a hero is not and empty string.

File: app/controllers/post_controller.rb (hunk 1 above)
Comment (andy-wojciechowski) [reply app/controllers/post_controller.rb]: Also, please add a test case here for when the hero is a valid URL. I think all of the tests currently use empty strings as a hero.
